### ISSUE TITLE
Support manual UX testing of bot changes before rolling out to the general public

### ISSUE THREAD

**Issue by huydhn** [pytorch/test-infra#464]
### Pitch

One issue that Zain and I briefly discussed with Kerry when learning about PyTorch bot is that any change in the bot will become available right away to everyone as soon as it's merged and deployed by Vercel. This works for smaller changes like https://github.com/pytorch/test-infra/issues/409. However, it's arguable that having a way to manually trigger the workflow change to test that everything works end-to-end before release it to the public is nicer. This also allows us to innovate on UX change without worrying about breaking everyone workflow.

An example user story:

I'm changing `merge -f` to provide a reason `merge -f -m 'FOOBAR'`. After the change is merged and deployed on `test-infra`, I want to test the change on my own branch or a test branch on https://github.com/pytorch/pytorch first to confirm that everything works via the UI. Once this is done, I can decide to rework some aspects or release the change. This is more important for bigger or more complicated changes.

### Proposed change

_May be we already have a way to achieve this. if that's the case, let me know so that I can document and pin it somewhere_

1. Design and implement a mechanism in pytorchbot to make it easier to test new UX changes in a test branch. I'm new to typescript, but I'm thinking of something like this:
```
@pytorchbot_experimentation(branch="foobar")
def new_feature(args):
    pass
```
2. Designate a dedicated branch on https://github.com/pytorch/pytorch to test UX changes. IMO, it's better to make this configurable so that people can use their own branches.
3. Once the change is ready, just need to submit another PR to remove the `@pytorchbot_experimentation` annotation.

Thoughts?

cc @ZainRizvi @zengk95 @janeyx99 @suo

**Comment by ZainRizvi** [comment]
+1 to the concept of being able to test our changes before they break other people's workflows by marking them as experimental :)

I'm not sure I fully grok what you mean by #2 though.

Two similar implementation options could be:

1. Setting up a test branch like you suggested and have Vercel deploy that test branch to a new endpoint. That would prob require us to configure an additional repository (like pytorch-canary that @zengk95 suggested) that the test endpoint will listen for mergebot commands on. 

2. Just deploy everything the same way we do today, but require an extra flag to mark when the experimental version of a command should be used (e.g. we could use `-x` for an effect like `@mergebot -x -f [BUILDBREAK] The build was broken"`)

**Comment by huydhn** [comment]
> I'm not sure I fully grok what you mean by #2 though.

Oh, it just means to be able to use a custom repo or branch for testing. There are definitely more than one way to achieve this. Personally, I would prefer to use by own fork repo and branch. Pytorch canary is also fine. Lastly, if we can also use pytorch/pytorch repo (for convenience) given that a custom branch can be set there. I'm not too strongly for or against any of these approaches I guess

**Comment by ZainRizvi** [comment]
Ah, gotcha. If there's a way to test it using a private branch then that would be ideal :D
